pkgs/build-support/binary-cache/make-binary-cache.py

```diff
@@ -1,43 +1,78 @@
-
+from functools import partial
 import json
+from multiprocessing import Pool
 import os
 import subprocess
 
-with open(os.environ["NIX_ATTRS_JSON_FILE"], "r") as f:
-  closures = json.load(f)["closure"]
 
-os.chdir(os.environ["out"])
+def dropPrefix(path, nixPrefix):
+    return path[len(nixPrefix + "/") :]
+
+
+def processItem(item, nixPrefix, outDir):
+    os.chdir(outDir)
+
+    narInfoHash = dropPrefix(item["path"], nixPrefix).split("-")[0]
+
+    xzFile = "nar/" + narInfoHash + ".nar.xz"
+    with open(xzFile, "w") as f:
+        subprocess.run(
+            "nix-store --dump %s | xz -c" % item["path"],
+            stdout=f,
+            shell=True,
+            check=True,
+        )
+
+    fileHash = (
+        subprocess.run(
+            ["nix-hash", "--base32", "--type", "sha256", "--flat", xzFile],
+            capture_output=True,
+            check=True,
+        )
+        .stdout.decode()
+        .strip()
+    )
+    fileSize = os.path.getsize(xzFile)
+
+    finalXzFile = "nar/" + fileHash + ".nar.xz"
+    os.rename(xzFile, finalXzFile)
+
+    with open(narInfoHash + ".narinfo", "w") as f:
```

Comment:
Text-mode is the default, but this shows the code means to do so.

```suggestion
    with open(f"{narInfoHash}.narinfo", "wt") as f:
```